autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -517,11 +525,17 @@ def infer_objects(self, X: pd.DataFrame) -> pd.DataFrame:
                     self.logger.warning(f'Casting the column {key} to {dtype} caused the exception {e}')
                     pass
         else:
-            # Calling for the first time to infer the categories
-            X = X.infer_objects()
-            for column, data_type in zip(X.columns, X.dtypes):
-                if not is_numeric_dtype(data_type):
-                    X[column] = X[column].astype('category')
+            if len(self.dtypes) != 0:
+                # when train data has no object dtype, but test does
+                # we prioritise the datatype given in training data
+                for column, data_type in zip(X.columns, self.dtypes):
+                    X[column] = X[column].astype(data_type)
```

Comment:
It does not work if the number of features (let's say 10^5) is huge.
So could you change it to the following?
```suggestion
                dtype_dict = {col: dtype for col, dtype in zip(X.columns, self.dtypes)}
                X = X.astype(dtype_dict)
```